File: task3-unknown-language/noteapp/internal/model/note.go
package model

import "time"

// Note представляет структуру заметки
type Note struct {
	ID        int       `json:"id" db:"id"`
	Text      string    `json:"text" db:"text"`
	CreatedAt time.Time `json:"created_at" db:"created_at"`
}

// CreateNoteRequest представляет запрос на создание заметки
type CreateNoteRequest struct {
	Text string `json:"text"`
}

// UpdateNoteRequest представляет запрос на обновление заметки
type UpdateNoteRequest struct {
	Text string `json:"text"`
}

// ValidateText проверяет валидность текста заметки
func ValidateText(text string) error {
	if text == "" {
		return ErrEmptyText
	}
	if len(text) > 500 {
		return ErrTextTooLong
	}
	return nil
}

File: task3-unknown-language/Kotlin-notes-app/src/main/kotlin/com/notes/model/Note.kt
package com.notes.model

import kotlinx.serialization.Serializable
import kotlinx.datetime.Instant

/**
 * Модель заметки
 * @property id Уникальный идентификатор заметки
 * @property text Текст заметки (1-500 символов)
 * @property createdAt Время создания заметки
 */
@Serializable
data class Note(
    val id: Long = 0,
    val text: String,
    val createdAt: String
)

/**
 * DTO для создания новой заметки
 * @property text Текст заметки
 */
@Serializable
data class CreateNoteRequest(
    val text: String
)

/**
 * DTO для обновления заметки
 * @property text Новый текст заметки
 */
@Serializable
data class UpdateNoteRequest(
    val text: String
)

/**
 * DTO для ответа с ошибкой
 * @property error Сообщение об ошибке
 */
@Serializable
data class ErrorResponse(
    val error: String
)